### ISSUE TITLE
ImportError: No module named utils using google colab

### ISSUE THREAD

**Issue by theman162** [tensorflow/models#8468]
Hi i am using google colab to run object_detection_tutorial.ipynb from https://github.com/EdjeElectronics/TensorFlow-Object-Detection-API-Tutorial-Train-Multiple-Objects-Windows-10.I am using tensorflow cpu and anaconda prompt.I have added the utils module to path in anaconda prompt
set PYTHONPATH=C:\tensorflow2\models;C:\tensorflow2\models\research;C:\tensorflow2\models\research\slim;C:\tensorflow2\models\research\object_detection\utils.Yet this issue is still continues

ImportErrorTraceback (most recent call last)
in ()
----> 1 from utils import label_map_util
2
3 from utils import visualization_utils as vis_util

ImportError: No module named utils

**Comment by ravikyram** [comment]
@theman162 

Can you try with below code instead and see if it helps you.Thanks!
```
from object_detection.utils import label_map_util
from object_detection.utils import visualization_utils as vis_util
```

**Comment by theman162** [comment]
![image](https://user-images.githubusercontent.com/57122013/81050220-332c6a00-8ef2-11ea-8661-f4b8997cd169.png)

**Comment by theman162** [comment]
OS Platform and Distribution :windows 10

TensorFlow version:1.5

Python version:python 3.7

Installed using virtualenv? pip? conda?:pip

GCC/Compiler version (if compiling from source):

CUDA/cuDNN version: using tensorflow cpu

GPU model and memory:using tensorflow cpu

**Comment by ravikyram** [comment]
@theman162 

Can you please refer the [link](https://stackoverflow.com/questions/49545142/modulenotfounderror-no-module-named-utils) and see if it helps you.Thanks!

**Comment by theman162** [comment]
I have already tried everything from that link it  does not help

**Comment by AshishGusain17** [comment]
@theman162 
Run  %ls and check in which directory you are.
Than use  %cd <path to research folder> andd you will be able to access object_detection folder

**Comment by pkulzc** [comment]
The repo you mentioned is not maintained by us so you probably want to ask your question there. They require different installations so it's hard to tell if you have finished all required installation steps. The first thing you want to check is if you are running your code in the correct path (aka the object_detection dir.

**Comment by AjibolaPy** [comment]
Please how do i use the utils.colab_stream_utils i saw it on yolo github page as a streaming feature how do i add it and activate it

**Comment by xfiax** [comment]
Having similar issue. the above answers don't solve the problem..

**Comment by YoungMoneyInvestments** [comment]
> Having similar issue. the above answers don't solve the problem..

The term "utils" in Python typically refers to a utility module or package. It's a common practice in programming to create a module named `utils` (short for "utilities") which contains various helper functions and classes that are used across different parts of a project. This module serves as a central place for code that is not specific to any single feature or domain but is shared across different areas of the application.

### Characteristics of a `utils` Module:

1. **General Purpose**: Functions and classes in a `utils` module are generally not specific to a particular application logic but are reusable in various contexts.

2. **Helper Functions**: They often include helper functions for common tasks like date and time manipulation, string processing, file operations, and more.

3. **Code Organization**: Having a `utils` module helps in organizing and managing code by reducing duplication and promoting reusability.

4. **Custom vs. Standard Libraries**: While Python and its extensive standard library provide many utility functions, developers often create custom `utils` modules for tasks unique to their specific project.

### Examples of Utils Functions:

- Date and time conversions (e.g., converting timestamps to human-readable formats).
- File reading and writing utilities.
- Data validation checks.
- String manipulation functions.
- Mathematical operations that are frequently used across the project.

### Note on Usage:

- **Specific vs. Generic**: While `utils` modules are helpful, it's important to avoid making them a "dumping ground" for all miscellaneous code. Functions and classes should be placed in `utils` only if they don't logically fit into more specific modules.

- **Third-Party Utils**: There are also third-party packages and libraries named `utils` or similar, which might provide specific functionalities. These should be used based on the needs of the project and understanding of what the package offers.

In summary, `utils` in Python is a convention used to name modules that contain utility functions and classes helpful across different parts of a software project. The exact contents and functions of a `utils` module can vary significantly depending on the specific requirements and context of the project.
